src/main/java/com/jcabi/jdbc/StoredProcedureOutcome.java

```
@@ -0,0 +1,110 @@
+ * @version $Id$
+ * @since 0.17
+ * @param <T> Type of items
+ */
+@Immutable
+@ToString
+@EqualsAndHashCode(of = "indexes")
+public final class StoredProcedureOutcome<T> implements Outcome<T> {
+
+    /**
+     * OUT parameters' indexes.
+     */
+    @Immutable.Array
+    private final transient int[] indexes;
+
+    /**
+     * Ctor.
+     * @param params Number of OUT params. Has to be > 0.
+     *  If this ctor is used, it is assumed that the OUT parameters
+     *  are from index 1 to including nrop.
+     */
+    public StoredProcedureOutcome(final int params) {
+        if (params <= 0) {
+            throw new IllegalArgumentException(
+                "Nr of out params has to be a positive int!"
+            );
+        }
+        this.indexes = new int[params];
+        for (int idx = 0; idx < params; ++idx) {
+            this.indexes[idx] = idx + 1;
+        }
+    }
+
+    /**
+     * Ctor.
+     * @param indexes Indexes of the OUT params.
+     *  <b>Index count starts from 1</b>.
+     */
+    public StoredProcedureOutcome(final int... indexes) {
+        if (indexes.length == 0) {
+            throw new IllegalArgumentException(
+                "At least 1 OUT param's index needs to be specified!"
+            );
+        }
+        final int size = indexes.length;
+        this.indexes = new int[size];
+        for (int idx = 0; idx < size; ++idx) {
+            this.indexes[idx] = indexes[idx];
+        }
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public T handle(
+        final ResultSet rset, final Statement stmt) throws SQLException {
+        final int params = this.indexes.length;
+        final Object[] outs = new Object[params];
+        if (stmt instanceof CallableStatement) {
+            for (int idx = 0; idx < params; ++idx) {
+                outs[idx] = ((CallableStatement) stmt).getObject(
+                    this.indexes[idx]
+                );
+            }
+        }
+        return (T) outs;
```

@mkordas Right - I see in other Outcomes they solved the issue by requesting from the user the ``Class<T>`` object in the constructor, and later in ``handle`` it compares it to different acceped ``.class``, like ``String``, ``byte[]`` etc. Finally if none matches, ``IllegalArgumentException`` 

In my case I would let it fail and explain the reason in the Javadoc, since asking for the ``Class<T>`` argument in the constructor seems pointless (it could only have the value ``Object[].class``, so why make it a parameter? It should be intuitive though, since you're calling a stored procedure and my ``Outcome`` cannot know which types are there, so you should expect an array of ``Object[]``